### DISCUSSION TITLE
How about a light version?

### DISCUSSION THREAD

**Discussion by rodrigosaling** [one-dark/jetbrains-one-dark-theme#206] [Ideas]
Hello!

I used this theme quite a lot, and it is hands down the best dark theme for JetBrains IDEs. But some months ago, I decided to travel back in time to the _Dreamweaver era_ and started looking for light themes because it felt better on my eyes (having to frequently change between a dark IDE and a bright website doesn't feel right).

Have you ever thought on building a light version of One Dark? I've never created a JetBrains theme, but I can help with that. I guess you have all the shortcuts and knowledge to create a new theme. 🙏

I am currently using the **Rider Light** color scheme from https://plugins.jetbrains.com/plugin/13883-rider-ui-theme-pack. And the **GitHub theme** for VS Code is really nice (https://marketplace.visualstudio.com/items?itemName=GitHub.github-vscode-theme). What I like about both themes is that they don't use underlines on variables, they use some gray-ish backgrounds, and have more colors than just blue, green, and black.

Thank you!

**Comment by mskelton** [comment]
Hi @rodrigosaling!

This was discussed twice in issues https://github.com/one-dark/jetbrains-one-dark-theme/issues/66 and https://github.com/one-dark/jetbrains-one-dark-theme/issues/95 the second of which led to an email thread with some recommendations I gave for developing a light variant of the theme.  The main point that I've commented in both issues is:

> While I'm not opposed to the idea of having the light version of this theme, I think it is so different from this theme that it should be in a separate repository/IntelliJ plugin. Because this theme is just a color theme and doesn't have any other customizations or features, most of this repo would have to change.

The second issue led to an email thread shown below in which the contributor created the following GitHub org/repo https://github.com/one-light/jetbrains-one-light-theme.  You could reach out about helping on that or just create your own repo.  Anyway, here is the email thread.

**Contributor**

> I created a suggestion on One Dark theme repo several days ago about Light version of the theme. I have no prior experience with development of such things but i think that i want to create intellij version of One Light since now there is none. Maybe you can suggest how can i transfer the colors of original Atom theme to intellij?

**Me**

> The biggest lesson I learned from making the One Dark theme is that having a solidly defined color palette and sticking to that color palette as much as possible makes the theme consistent and simple to maintain.
> 
> The color palette for one dark is here: https://one-dark.gitbook.io/jetbrains/contributing/development#color-palette.
> 
> One Light will obviously have a very different color palette and I encourage you to create some fun names for each of the colors (disclaimer: I didn't come up with the names for One Dark but got them from a different repo :)).  Looks like you may be able to snag the colors from Atom's GitHub repo here (https://github.com/atom/atom/blob/master/packages/one-light-syntax/styles/colors.less).
> 
> After you have your color palette, the next biggest challenge is getting a repo setup and starting to update all the colors.  I would be more than happy to help set that up, in fact I think it would give me a chance to rewrite the build scripts and put them into a npm package so they can be shared by both repos.  Additionally, I would suggest snagging the one-light organization name in GitHub (https://github.com/one-light) so you can put the theme there.  That way other one light themes that you or others develop could be grouped together.

**Contributor**

> I'm started to playing around with theming intellij to match One Light color scheme and i have one question to you. I get the main base colors from atom repo and i will use them on editor scheme, but the rest of the intellij have a lot more colors than that and i can see that also in your theme json file. Maybe you have some suggestions on how to pick other colors for menus, active tabs and other stuff?

**Me**

> The editor colors (one_dark.theme.json) were a lot of manual work of figuring out what looked good.  There isn't really a special formula for it but I do have some recommendations of things that helped when I did it for the dark theme.
> 1. I would recommend focusing first on the code color scheme as that will be more important to have colored nicely than the editor itself.  You can have the editor inherit from the default light theme which will at least make the window a light color.
> 1. Once you get the code color scheme in a good place, start working through the editor colors section by section.  I started with a blank slate and then added chunks to the one_dark.theme.json file.
> 1. One of the most difficult things was finding what some of the settings in one_dark.theme.json were applied to.  To make it easier to spot, I set the values to an obnoxious hot pink color so that when I found the elements that the styles applied to I couldn't miss it.
> 1. A bunch of UI elements are available when you have [internal actions](https://plugins.jetbrains.com/docs/intellij/enabling-internal.html?from=jetbrains.org) turned on and you execute the command "Show Standard Panels".  This way you can easily see how elements will look with your theme without having to find how to make those UI elements show up.

**Reply by rodrigosaling** [reply]
Thank you for such detailed answer!

**Reply by akulik512** [reply]
Hello @rodrigosaling! 
Do you have any news in your research about the "light" fork for the One Dark?
I try to migrate the "One Dark Vivid Italic" just change some colours in `.icls` but it's pretty difficult and I don't understand what I do :smile: 
I hope you succeeded more.

**Reply by rodrigosaling** [reply]
Hi @akulik512 ! Sorry but I didn't do anything related to this 😔 I didn't even try to do a migration or a fork as you did.
Maybe in the future 😓
